### PR TITLE
chore: add mvi example snippets for filter expressions

### PR DESCRIPTION
**Pull request by malandis**
This PR adds an example snippet demonstrating vector index filter expressions. Because each filter expression type is (in my opinion) straigthforward on its own, I add a holistic example showing many expressions over a single metadata schema.

### CODE REVIEW

File: examples/nodejs/vector-index/doc-example-files/doc-examples-nodejs-apis.ts [hunk 1]
```diff
@@ -141,6 +142,66 @@ async function example_API_SearchAndFetchVectors(vectorClient: PreviewVectorInde
...
+  // Was the movie released in 1999?
+  F.equals('year', 1999);
+
+  // Did the movie gross 463.5 million dollars?
+  F.equals('gross_revenue_millions', 463.5);
+
+  // Was the movie in theaters?
+  F.equals('in_theaters', true);
+
+  // Was the movie not in theaters?
```
Comment (nand4011) [inline examples/nodejs/vector-index/doc-example-files/doc-examples-nodejs-apis.ts:177]: Was the movie released after 1990?

File: examples/nodejs/vector-index/doc-example-files/doc-examples-nodejs-apis.ts [hunk 2]
```diff
@@ -141,6 +142,66 @@ async function example_API_SearchAndFetchVectors(vectorClient: PreviewVectorInde
...
+  F.greaterThan('year', 1990);
+
+  // Was the movie released after 2020?
```
Comment (nand4011) [inline examples/nodejs/vector-index/doc-example-files/doc-examples-nodejs-apis.ts:180]: Was the movie release in or after 2020?

**Review by nand4011 — COMMENTED** [review]
A couple of comments need to be updated in the examples, but lgtm otherwise.

File: examples/nodejs/vector-index/doc-example-files/doc-examples-nodejs-apis.ts (hunk 2 above)
Comment (malandis) [reply examples/nodejs/vector-index/doc-example-files/doc-examples-nodejs-apis.ts]: Fixed in https://github.com/momentohq/client-sdk-javascript/pull/1130/commits/5519040aadeaba4bcd2168243bf57988c28d0c61

**Review by nand4011 — APPROVED** [review]

**Review by pgautier404 — APPROVED** [review]
These are really cool :-)

**Pull request merged**